src/platform/terminals/codeExecution/codeExecutionHelper.node.ts

```diff
@@ -31,12 +31,17 @@ export class CodeExecutionHelper extends CodeExecutionHelperBase {
 
     public override async normalizeLines(code: string, resource?: Uri): Promise<string> {
         try {
-            if (code.trim().length === 0) {
+            const codeTrimmed = code.trim();
+            if (codeTrimmed.length === 0) {
                 return '';
             }
             // On windows cr is not handled well by python when passing in/out via stdin/stdout.
             // So just remove cr from the input.
             code = code.replace(new RegExp('\\r', 'g'), '');
+            if (codeTrimmed.indexOf('\n') === -1) {
+                // the input is a single line, maybe indented, without terminator
+                return codeTrimmed + '\n';
```

Comment:
I believe this is identical to the result from launching the subproecss except the case `if True:`, but that's fine because we'll get an error in the interactive either way (could be `if True:\n` or `if True:\n\n`)